kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/splash/presenter/SplashPresenter.java

```diff
@@ -31,65 +76,112 @@ public void onAttach(ISplashView view) {
 	}
 
 	@Override
-    public void getStartedClicked() {
-    	eventLogger.send(WelcomeScreenButtonTapped.create());
-        animateLoading();
-        activateAccount();
-    }
-
-    private void animateLoading() {
-        if (view != null) {
-            view.animateLoading();
-        }
-    }
-
-    private void activateAccount() {
-        authRepository.activateAccount(new KinCallback<Void>() {
-            @Override
-            public void onResponse(Void response) {
-                confirmedSucceed = true;
-                navigateToMarketplace();
-            }
-
-            @Override
-            public void onFailure(KinEcosystemException exception) {
-                showToast("Oops something went wrong...");
-                stopLoading(true);
-            }
-        });
-    }
-
-    private void stopLoading(boolean reset) {
-        if (view != null) {
-            view.stopLoading(reset);
-        }
-    }
-
-    private void showToast(String msg) {
-        if (view != null) {
-            view.showToast(msg);
-        }
-    }
-
-    private void navigateToMarketplace() {
-        if (animationEnded && confirmedSucceed) {
-            if (view != null) {
-                view.navigateToMarketPlace();
-            }
-        }
-    }
-
-    @Override
-    public void backButtonPressed() {
-        eventLogger.send(BackButtonOnWelcomeScreenPageTapped.create());
-        if (view != null) {
-            view.navigateBack();
-        }
-    }
-
-    @Override
-    public void onAnimationEnded() {
-        animationEnded = true;
-        navigateToMarketplace();
-    }
+	public void onDetach() {
+		super.onDetach();
+		removeAccountStateObserver();
+		cancelTimeoutTask();
```

Comment:
what happen if onDetach is called an then onAttached() isCalled again the timeout will not run and user can wait more than 20 sec?

Reply:
onDetach is being called from activity onDestroy so no option for the scenario you just mentioned.